backend/src/federation/index.ts

```diff
@@ -36,18 +32,60 @@ export const startDHT = async (
     logger.info(`keyPairDHT: publicKey=${keyPair.publicKey.toString('hex')}`)
     logger.debug(`keyPairDHT: secretKey=${keyPair.secretKey.toString('hex')}`)
 
+    const apiList = {
+      apiVersions: Object.values(ApiVersionType).map(function (apiEnum) {
+        return { api: apiEnum, url: CONFIG.FEDERATION_COMMUNITY_URL }
+      }),
+    }
+    logger.debug(`ApiList: ${JSON.stringify(apiList)}`)
+
     const node = new DHT({ keyPair })
 
     const server = node.createServer()
 
     server.on('connection', function (socket: any) {
       // noiseSocket is E2E between you and the other peer
       // pipe it somewhere like any duplex stream
-      logger.info(`Remote public key: ${socket.remotePublicKey.toString('hex')}`)
+      logger.info(`server on... with Remote public key: ${socket.remotePublicKey.toString('hex')}`)
       // console.log("Local public key", noiseSocket.publicKey.toString("hex")); // same as keyPair.publicKey
 
-      socket.on('data', (data: Buffer) => logger.info(`data: ${data.toString('ascii')}`))
-
+      socket.on('data', async (data: Buffer) => {
+        try {
+          logger.info(`data: ${data.toString('ascii')}`)
+          const json = JSON.parse(data.toString('ascii'))
+          if (
+            json.apiVersions &&
+            Object.prototype.toString.call(json.apiVersions) === '[object Array]' &&
+            json.apiVersions.length > 0 &&
+            Object.prototype.toString.call(json.apiVersions[0].api) === '[object String]' &&
```

Comment:
Why not use `typeof var`?
https://stackoverflow.com/questions/4059147/check-if-a-variable-is-a-string-in-javascript

Reply:
From some answers out of the foren it was not clear what the typeof operator will return for a String: 'object' or as expected 'string' 
The Object.prototype.toString.call solution seems to be working correct and returns always [object string] for both 'string' and 'String'.
I've tried it now and changed it to the "typeof var" solution